### PR TITLE
Clarify cause of warnings about met data time coverage

### LINKED ISSUE #153: UserWarning: Met data 'rad' does not overlap the grid domain along the time axis. Lead to missing values in first hour of eval.
**Description**

Loading met and rad data from cachestore with same start and end times leads to UserWarning only for rad parameters. Consequently, the first hour of eval does not contain any values. I think this is not the intended behavior, since ::func:_check_met_rad_times (lines 141-153 in cocip_time_handling) checks rad and met with same tmin and tmax values.

**Details**

- Version: v0.49.3 (but also v0.49.2/1)
- Module: models.cocipgrid.cocip_time_handling
- OS: linux cluster
- Reporter: kirscsim

**Steps to Reproduce**

Minimal example:

```import numpy as np

from pycontrails import DiskCacheStore
from pycontrails.datalib.ecmwf import ERA5, HRES
from pycontrails.models.cocip import Cocip
from pycontrails.models.cocipgrid import CocipGrid
from pycontrails.models.ps_model.ps_grid import PSGrid
from pycontrails.core.met import MetDataset
from pycontrails.models.humidity_scaling import ConstantHumidityScaling
from datetime import datetime, timedelta
```

Data loading from cachestore and preprocessing:

```
t0 = datetime(2024,2,5,0)
t1 = datetime(2024,2,5,13)

pressure_levels = [
    900, 850, 800, 700, 600, 500, 400, 300, 250, 200, 150, 100, 70, 50, 30
]

forecast_time = datetime(2024,2,5,0)

cachestore = DiskCacheStore(
        cache_dir='path/to/cachestore', 
        allow_clear=False,
    )

met = HRES(
        time=(t0, t1),
        variables=Cocip.met_variables + Cocip.optional_met_variables, 
        pressure_levels=pressure_levels, 
        grid=.25,
        cachestore=cachestore,
        url='https://api.ecmwf.int/v1',
        key=***,
        email=***,
        forecast_time=forecast_time,
)
    
rad = HRES(
    time=(t0, t1),
    variables=Cocip.rad_variables,
    grid=.25,
    cachestore=cachestore, 
    url='https://api.ecmwf.int/v1',
    key=***,
    email=***,
    forecast_time=forecast_time,
)

met = met.open_metdataset(xr_kwargs=dict(parallel=False))
rad = rad.open_metdataset(xr_kwargs=dict(parallel=False))

met = met.wrap_longitude()
rad = rad.wrap_longitude()
```

Met times:
```met['time'].values.min(), met['time'].values.max()```
output:
```
(numpy.datetime64('2024-02-05T00:00:00.000000000'),
 numpy.datetime64('2024-02-05T13:00:00.000000000'))
```

Rad times:
```rad['time'].values.min(), rad['time'].values.max()```
output:
```
(numpy.datetime64('2024-02-05T00:00:00.000000000'),
 numpy.datetime64('2024-02-05T13:00:00.000000000'))
```

Cocipgrid calculation:
```
cocipgrid = CocipGrid(
    met=met,
    rad=rad,
    process_emissions=True,
    downselect_met=False,
    humidity_scaling=ConstantHumidityScaling(rhi_adj=0.99),
    aircraft_performance=PSGrid(),
)

# Define the 4D grid box that should be evaluated
delta = 0.25
longitude = np.arange(-20, 20, delta)
latitude = np.arange(50, 70, delta)
level = [
    376.0, 359.0, 344.0, 329.0, 314.0, 300.0, 287.0, 274.0, 262.0, 250.0, 
    238.0, 227.0, 216.0, 206.0, 196.0, 187.0,
]
time = [t0 + timedelta(hours=i) for i in range(2)]
print(time)
grid = MetDataset.from_coords(longitude, latitude, level, time)

# Apply the gridded CoCiPcd
result = cocipgrid.eval(
    source=grid,
    max_age=np.timedelta64(12, 'h'),
)
```

Output:
```
[datetime.datetime(2024, 2, 5, 0, 0), datetime.datetime(2024, 2, 5, 1, 0)]

***/micromamba/envs/pycontrails_complete/lib/python3.11/site-packages/pycontrails/models/cocipgrid/cocip_grid.py:2193: UserWarning: Met data 'rad' does not overlap the grid domain along the time axis. This causes interpolated values to be nan, leading to meaningless results.
  warnings.warn(
***/micromamba/envs/pycontrails_complete/lib/python3.11/site-packages/pycontrails/models/cocipgrid/cocip_time_handling.py:149: UserWarning: Parameter 'rad' is too short in the time dimension. Include additional time in 'rad' or reduce 'max_age' parameter.Model start time: 2024-02-05 00:00:00 Model end time: 2024-02-05 13:00:00
  warnings.warn(

CocipGrid eval: 77%
43/56 [01:21<00:25, 1.95s/it]
```

**Additional Notes**

I expect a warning from ::func:_check_met_rad_times from both met and rad or None. I also tried to increase time by adding time at the end, which resulted in the same behavior. Adding time before the first hour solved this issue and is my current workaround.

Here visualization:
Issue_fig0
Issue_fig1

### PR DESCRIPTION
**Pull request by thabbott**
Closes #153 

## Changes

Improves the clarity of warnings produced when meteorology data doesn't cover the time range required by a gridded Cocip model.

Example:
```python
model = CocipGrid(
    met=met,   # HRES data starting at 2024-02-19T00
    rad=rad,   # HRES data starting at 2024-02-19T00
    max_age=np.timedelta64(10, "h"),
    ...
)

# Gridded model that also starts at 2024-02-19T00
source = CocipGrid.create_source(level=5, time=np.datetime64("2024-02-19T00"))
_ = model.eval(source)
```

Previous warnings (time-related only):
```
UserWarning: Met data 'rad' does not overlap the grid domain along the time axis. This causes interpolated values to be nan, leading to meaningless results.
UserWarning: Parameter 'rad' is too short in the time dimension. Include additional time in 'rad' or reduce 'max_age' parameter.Model start time: 2024-02-19 00:00:00 Model end time: 2024-02-19 10:00:00
```

New warnings (time-related only):
```
UserWarning: Met data 'rad' does not overlap the grid domain along the time axis. This causes interpolated values to be nan, leading to meaningless results.
UserWarning: Start time of parameter 'rad' (2024-02-19 00:30:00) is after model start time (2024-02-19 00:00:00). Include additional time at the start of 'rad'. Note: differencing reduces time coverage when providing accumulated radiative fluxes.
```

#### Internals

- Modify unit tests to test updated warnings

## Tests

- [x] QC test passes locally (`make test`)
- [x] CI tests pass

## Reviewer

> @zebengberg

### CODE REVIEW

File: pycontrails/models/cocipgrid/cocip_time_handling.py [hunk 1]
```diff
@@ -340,3 +348,29 @@ class CocipRuntimeStats:
...
+        warnings.warn(
+            f"Start time of parameter '{name}' ({met_start}) "
+            f"is after model start time ({model_start}). "
+            f"Include additional time at the start of '{name}'."
+            f"{note}"
+        )
+
+
+def _check_end_time(
+    met_end: pd.Timestap, model_end: pd.TimeStamp, name: str, note: str | None = None
```
Comment (zebengberg) [inline pycontrails/models/cocipgrid/cocip_time_handling.py:367]: typo: Timestap -> Timestamp

Comment (zebengberg) [inline pycontrails/models/cocipgrid/cocip_time_handling.py:367]: You may want to make `note` a keyword-only variable both here and in `_check_end_time`

`... name: str, note: str | None ...` -> `... name: str, *, note: str | None ...`

File: pycontrails/models/cocipgrid/cocip_time_handling.py [hunk 2]
```diff
@@ -139,18 +139,26 @@ def attach_timedict(self) -> None:
...
+        ----------
+        tmin: pd.Timestamp
+            Start of required time range
+        tmax:pd.Timestamp
+            End of required time range
+        """
+        met_tmin = pd.to_datetime(self.met.data["time"].min().values)
+        met_tmax = pd.to_datetime(self.met.data["time"].max().values)
+        rad_tmin = pd.to_datetime(self.rad.data["time"].min().values)
+        rad_tmax = pd.to_datetime(self.rad.data["time"].max().values)
```
Comment (zebengberg) [inline pycontrails/models/cocipgrid/cocip_time_handling.py:154]: It might be a hair more performant to do something like:

```python
met_time = met.data["time"].values
met_tmin = pd.to_datetime(met_time.min())
met_tmax = pd.to_datetime(met_time.max())
...
```

**Review by zebengberg — APPROVED** [review]
Thanks Tristan!

I left a few optional suggestions -- feel free to ignore.

File: pycontrails/models/cocipgrid/cocip_time_handling.py (hunk 1 above)
Comment (thabbott) [reply pycontrails/models/cocipgrid/cocip_time_handling.py:367]: Good catch! (`TimeStamp` is also wrong.) Do you understand why mypy didn't catch this?
